optimize/functions/functions.go

```diff
@@ -60,14 +60,13 @@ func (Beale) Grad(grad, x []float64) []float64 {
 	return grad
 }
 
-func (Beale) Hess(hess mat.Symmetric, x []float64) mat.Symmetric {
+func (Beale) Hess(hess *mat.SymDense, x []float64) {
```

Comment:
Should these be `dst` instead of `hess`?

Reply:
Yes. I think they should.

Done